packages/devtools-reps/src/object-inspector/tests/component/entries.js

```diff
@@ -94,66 +102,84 @@ describe("ObjectInspector - entries", () => {
     expect(protoNode.find(".arrow").exists()).toBeTruthy();
     expect(protoNode.text()).toMatch(/^__proto__/);
 
-    // loadEntries shouldn't have been called since everything
+    // enumEntries shouldn't have been called since everything
     // is already in the preview property.
-    expect(loadObjectEntries.mock.calls.length).toBe(0);
+    expect(enumEntries.mock.calls.length).toBe(0);
   });
 
-  it("does not load entries if getObjectEntries returns a truthy element", () => {
-    const stub = gripMapRepStubs.get("testSymbolKeyedMap");
-    const loadObjectEntries = jest.fn();
-    const loadObjectProperties = jest.fn();
+  it("does not call enumEntries if entries are already loaded", () => {
+    const stub = gripMapRepStubs.get("testMoreThanMaxEntries");
+    const enumEntries = jest.fn();
 
-    mount(ObjectInspector({
-      autoExpandDepth: 2,
+    const wrapper = mount(ObjectInspector(generateDefaults({
+      autoExpandDepth: 3,
       roots: [{
         path: "root",
         contents: {
           value: stub
         }
       }],
-      getObjectEntries: () => {},
-      getObjectProperties: actor => {
-        return {
-          ownProperties: stub.preview.entries,
-        };
-      },
-      loadObjectEntries,
-      loadObjectProperties,
-    }));
+      createObjectClient: (grip) => ObjectClient(grip, {enumEntries}),
+      loadedProperties: new Map([
+        ["root", mapStubs.get("properties")],
+        [`root/${SAFE_PATH_PREFIX}entries`, mapStubs.get("11-entries")],
+        ["root/__proto__", true]
+      ])
+    })));
+
+    const nodes = wrapper.find(".node");
+    /*
+     * The OI should look like:
+     * ▶︎ Map
+     *     size : 11
+     *   ▶︎ <entries>
+     *     ▶︎ 0: "key-0" → "value-0"
+     *       ▶︎ <key>: "key-0"
+     *       ▶︎ <value>: "value-0"
+     *     ▶︎ 1: "key-1" → "value-1"
+     *       ▶︎ <key>: "key-1"
+     *       ▶︎ <value>: "value-1"
+     *     ▶︎ 2: "key-2" → "value-2"
+     *       ▶︎ <key>: "key-2"
+     *       ▶︎ <value>: "value-2"
+     *     […]
+     *     ▶︎ 10: "key-10" → "value-10"
+     *       ▶︎ <key>: "key-10"
+     *       ▶︎ <value>: "value-10"
+     *   ▶︎ __proto__
+     */
+    expect(nodes.length).toBe(
```

Comment:
this might be a nice snapshot. The trick is for the snapshot to look like your drawing above, which is easy to do w/ a format function

`expect(formatTree(nodes)).toMatchSnapshot()`